src/Microsoft.AspNet.OData.Shared/IEdmDeltaDeletedEntityObjectOfT.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an instance of an <see cref="IEdmChangedObject"/>.
+    /// Holds the properties necessary to create the ODataDeltaDeletedEntry.
+    /// </summary>
+    public interface IEdmDeltaDeletedEntityObject<TStructuralType> : IEdmChangedObject<TStructuralType>
```

Comment:
this should just be:

     public interface IEdmDeltaDeletedEntityObject<TStructuralType> : IEdmChangedObject<TStructuralType>, IEdmDeltaDeletedEntityObject
    {
    } 

and not add any members (Id and Reason are just inherited from IEdmDeltaDeletedEntityObject)